code/modules/fissionreactor/fissionreactor.dm

```diff
@@ -0,0 +1,377 @@
+	var/list/coolant_ports=list()
+	var/list/casing_parts=list()
+	var/heat_capacity=0
+	var/fuel_reactivity=1
+	var/fuel_rods_affected_by_rods=0
+	
+	var/coolantport_counter=0 // this varible exists to ensure that all coolant ports get treated equally, because if we didn't it would have a flow prefrence towards the ports with lower indexes.
+	var/control_rod_insertion=1  //phase 1 vars. modified during runtime
+	var/temperature=0 //this is set last
+	
+	var/datum/gas_mixture/coolant
+
+	
+	var/zlevel=0 //positional varibles
+	var/origin_x=0
+	var/origin_y=0
+	var/corner_x=0 //uses corner calculations. this is for the sake of being easier to calculate.
+	var/corner_y=0
+	var/datum/fission_fuel/fuel=null
+
+	
+/datum/fission_reactor_holder/New()
+	..()
+	coolant = new /datum/gas_mixture
+	coolant.temperature = T20C //vaguely room temp.
+	coolant.volume = 2500
+
+/datum/fission_reactor_holder/proc/init_resize(var/turf/origin) //code responsible for setting up the parameters of the reactor.
+	if(!origin) //something has gone wrong.
+		return
+	
+	var/turf/wall_up=locate(origin.x,origin.y+1,origin.z)
+	var/turf/wall_down=locate(origin.x,origin.y-1,origin.z)
+	var/turf/wall_left=locate(origin.x-1,origin.y,origin.z)
+	var/turf/wall_right=locate(origin.x+1,origin.y,origin.z)
+	
+	var/directions=0
+	
+	//copy
+	var/list/wc = wall_up.contents
+	for (var/i=1,i<wc.len,i++)
+		if(istype(wc[i],/obj/machinery/fissionreactor) || /obj/machinery/atmospherics/unary/fissionreactor_coolantport )
+			directions|=NORTH
+			break
+
+	wc = wall_down.contents
+	for (var/i=1,i<wc.len,i++)
+		if(istype(wc[i],/obj/machinery/fissionreactor) || /obj/machinery/atmospherics/unary/fissionreactor_coolantport )
+			directions|=SOUTH
+			break
+
+	wc = wall_left.contents
+	for (var/i=1,i<wc.len,i++)
+		if(istype(wc[i],/obj/machinery/fissionreactor) || /obj/machinery/atmospherics/unary/fissionreactor_coolantport )
+			directions|=WEST
+			break
+
+	wc = wall_right.contents
+	for (var/i=1,i<wc.len,i++)
+		if(istype(wc[i],/obj/machinery/fissionreactor) || /obj/machinery/atmospherics/unary/fissionreactor_coolantport )
+			directions|=EAST
+			break
+	//paste
+	
+	//abort if we have an invalid placment (not at a corner)
+	if ( (directions & NORTH && directions & SOUTH) || (directions & EAST && directions & WEST)) //if there are walls on north+south/east+west, it is not in the right spot
+		return
+	if ( !(directions & (NORTH | SOUTH) ) || !(directions & (EAST | WEST) ) ) //if there is not a wall at north/south + east/west, it is not in the right spot
+		return
+		
+	var/xs=0
+	var/ys=0
+	
+	//get the lengths of the reactor.
+	if(directions&WEST) //x-
+		xs=-1
+		while(TRUE) //it'll be fiiiiiiiine.	
+			var/turf/turftosearch=locate(origin.x+xs-1,origin.y,origin.z)
+			var/list/contents = turftosearch.contents
+			for (var/i=1,i<contents.len,i++)
+				if(istype(contents[i],/obj/machinery/fissionreactor) || /obj/machinery/atmospherics/unary/fissionreactor_coolantport )
```

Comment:
i'm guessing the second part is supposed to be in an istype() too?
you could probably do a `#define isnuke(A) (istype(A,/obj/machinery/fissionreactor) || istype(A,/obj/machinery/atmospherics/unary/fissionreactor_coolantport))` on top of the file and then replace these long ass blocks with it

Reply:
aren't defines global?

Reply:
they are unless you #undef them afterwards
lots of files do that to reduce copypaste blocks